arangod/Cluster/TakeoverShardLeadership.cpp

```diff
@@ -71,25 +71,20 @@ std::string stripServerPrefix(std::string const& destination) {
 
 TakeoverShardLeadership::TakeoverShardLeadership(MaintenanceFeature& feature,
                                                  ActionDescription const& desc)
-    : ActionBase(feature, desc) {
+    : ActionBase(feature, desc),
+      ShardDefinition(desc.get(DATABASE), desc.get(SHARD)) {
   std::stringstream error;
 
   _labels.emplace(FAST_TRACK);
 
-  if (!desc.has(DATABASE)) {
-    error << "database must be specified";
-  }
-  TRI_ASSERT(desc.has(DATABASE));
-
   if (!desc.has(COLLECTION)) {
     error << "collection must be specified. ";
   }
   TRI_ASSERT(desc.has(COLLECTION));
-
-  if (!desc.has(SHARD)) {
-    error << "shard must be specified";
+  
+  if (!ShardDefinition::isValid()) {
+    error << "database and shard must be specified. ";
```

Comment:
```suggestion
    error << "database and shard must be specified.";
```

Reply:
The whitespace at the end is actually intentional, because error messages can be concatenated. This is the case for all maintenace action constructors, which simply do
```
error << "new error message. ";
```
The idea of the ". " at the end is to make concatenated messages easier to read. It is not my invention, it is already like that in all ctors already.